### PR TITLE
adding utility functions to parse model config parameters field

### PR DESCRIPTION
**Pull request by askhade**

### CODE REVIEW

File: include/triton/backend/backend_common.h [hunk 1]
```diff
@@ -546,4 +554,48 @@ TRITONSERVER_Error* GetParameterValue(
...
+    std::string* value, const std::string& default_value);
+
+/// Try to parse the requested parameter.
+///
+/// \param params The param in model config
+/// \param mkey Key in the model config.
+/// \param value The parsed int value.
+/// \param default_value Default value to use when key is not found.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_Error* TryParseParameter(
```
Comment (deadeyegoodwin) [inline include/triton/backend/backend_common.h:575]: In the protocols parameters can only have string, int64 or bool value, so change this to int64_t and remove the size_t version below.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: include/triton/backend/backend_common.h (hunk 1 above)
Comment (askhade) [reply include/triton/backend/backend_common.h:575]: message ModelParameter
{
  //@@  .. cpp:var:: string string_value
  //@@
  //@@     The string value of the parameter.
  //@@
  string string_value = 1;
}

map<string, ModelParameter> parameters = 14;  (https://github.com/triton-inference-server/common/blob/main/protobuf/model_config.proto#L1775)

string_value is string which can then be parsed as uint64, int, bool, int64 etc 

where is this enforced "In the protocols, parameters can only have string, int64 or bool value"

I dont understand why we cant have unit64. I am using this for setting gpu_mem_limit and it needs to be uint64

Comment (deadeyegoodwin) [reply include/triton/backend/backend_common.h:575]: My fault, I thought it was different kind of parameter. Can you change names to TryParseModelStringParameter

Comment (askhade) [reply include/triton/backend/backend_common.h]: Done

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**